server/methods/channels/add.js

```diff
@@ -13,6 +14,16 @@ Meteor.methods({
       throw new Meteor.Error(404, 'Team does not exist');
     }
 
+    // Insert direct channel
+    if (options && options.direct) {
+      name = _.sortBy(options.allowedUsers, function(s) {return s;}).join("+");
+      if (!Channels.findOne({teamId: teamId, name: name})) {
+        return Channels.insert({name: name, teamId: teamId, direct: true, allowedUsers: options.allowedUsers});
+      } else {
+        return Channels.update({name: name}, {$set: {name: name, teamId: teamId, direct: true, allowedUsers: options.allowedUsers}});
+      }
+    }
+
     // Get rid of extra spaces in names, lower-case it
     // (like Slack does), and trim it
     channelName = channelName.replace(/\s{2,}/g, ' ').toLowerCase().trim();
```

Comment:
@anbuselvan, can you do something that doesn't conflict too much with this PR, like, just simply modify the `check` statement:

```js
check(channelName, Match.Where(function (name) {
  check(name, String);
  return name.trim().length > 0;
});
```

Or something like that.